core/src/main/java/org/apache/gravitino/authorization/PermissionManager.java

```diff
@@ -373,11 +376,174 @@ User revokeRolesFromUser(String metalake, List<String> roles, String user) {
     }
   }
 
-  private List<String> toRoleNames(List<RoleEntity> roleEntities) {
-    return roleEntities.stream().map(RoleEntity::name).collect(Collectors.toList());
+  Role grantPrivilegeToRole(
+      String metalake, String role, MetadataObject object, List<Privilege> privileges) {
```

Comment:
I think you need to check if these privileges can be bound to this metadata object before you update the role.

Reply:
Also I think you need to dedup the privileges if user inputs the same privileges.

Reply:
I deduplicate privileges in the PermissonManager.

Reply:
What if two same privileges with one `allow` and one `deny` are both set, what will be happened and do you handle such scenario?

Reply:
I add new check logic to avoid this case.